backend-api/src/functions/proxy/proxyHandler.ts

```diff
@@ -0,0 +1,142 @@
+import {
+  APIGatewayProxyEvent,
+  APIGatewayProxyEventHeaders,
+  APIGatewayProxyResult,
+  Context,
+} from "aws-lambda";
+
+import { dependencies, IMockProxyDependencies } from "./handlerDependencies";
+import { ConfigService } from "./configService/configService";
+
+export type StandardisedHeaders = {
+  [key in string]: string | number | boolean;
+};
+
+export async function lambdaHandlerConstructor(
+  dependencies: IMockProxyDependencies,
+  event: APIGatewayProxyEvent,
+  context: Context,
+): Promise<APIGatewayProxyResult> {
+  const logger = dependencies.logger();
+  logger.addContext(context);
+
+  const configService = new ConfigService();
+  const configResult = configService.getConfig(dependencies.env);
+  logger.log("STARTED");
+  if (configResult.isError) {
+    logger.log("ENVIRONMENT_VARIABLE_MISSING", {
+      errorMessage: configResult.value.errorMessage,
+    });
+    return {
+      headers: {
+        "Content-Type": "application/json",
+      },
+      statusCode: 500,
+      body: JSON.stringify({
+        error: "server_error",
+        error_description: "Server Error",
+      }),
+    };
+  }
+
+  const { path } = event;
+  const allowedPaths = ["/async/token", "/async/credential"];
+
+  if (!allowedPaths.includes(path)) {
+    logger.log("UNEXPECTED_PATH", {
+      errorMessage: "Path is not one of the permitted values",
+    });
+    return {
+      headers: {
+        "Content-Type": "application/json",
+      },
+      statusCode: 500,
+      body: JSON.stringify({
+        error: "server_error",
+        error_description: "Server Error",
+      }),
+    };
+  }
+
+  const method = event.httpMethod;
+
+  if (method !== "POST") {
+    logger.log("UNEXPECTED_HTTP_METHOD", {
+      errorMessage: "Method is not an accepted value",
```

Comment:
If we only accept 'POST' here, is it worth making this message more specific? Super minor thought happy for you to dismiss